src/Aether/Devices/Drivers/Sgp4xAlgorithm.cs

```diff
@@ -0,0 +1,900 @@
+        const double VocAlgorithm_INIT_TRANSITION_VARIANCE = 0.01;
+        const double VocAlgorithm_GATING_THRESHOLD = 340;
+        const double VocAlgorithm_GATING_THRESHOLD_INITIAL = 510;
+        const double VocAlgorithm_GATING_THRESHOLD_TRANSITION = 0.09;
+        const double VocAlgorithm_GATING_MAX_DURATION_MINUTES = 60 * 3;
+        const double VocAlgorithm_GATING_MAX_RATIO = 0.3;
+        const double VocAlgorithm_SIGMOID_L = 500;
+        const double VocAlgorithm_SIGMOID_K = -0.0065;
+        const double VocAlgorithm_SIGMOID_X0 = 213;
+        const double VocAlgorithm_VOC_INDEX_OFFSET_DEFAULT = 100;
+        const double VocAlgorithm_LP_TAU_FAST = 20;
+        const double VocAlgorithm_LP_TAU_SLOW = 500;
+        const double VocAlgorithm_LP_ALPHA = -0.2;
+        const double VocAlgorithm_PERSISTENCE_UPTIME_GAMMA = 3 * 3600;
+        const double VocAlgorithm_MEAN_VARIANCE_ESTIMATOR__GAMMA_SCALING = 64;
+        const double VocAlgorithm_MEAN_VARIANCE_ESTIMATOR__FIX16_MAX = 32767;
+
+
+        internal struct VocAlgorithmParams
+        {
+            public int mVoc_Index_Offset;
+            public int mTau_Mean_Variance_Hours;
+            public int mGating_Max_Duration_Minutes;
+            public int mSraw_Std_Initial;
+            public int mUptime;
+            public int mSraw;
+            public int mVoc_Index;
+            public int m_Mean_Variance_Estimator__Gating_Max_Duration_Minutes;
+            public bool m_Mean_Variance_Estimator___Initialized;
+            public int m_Mean_Variance_Estimator___Mean;
+            public int m_Mean_Variance_Estimator___Sraw_Offset;
+            public int m_Mean_Variance_Estimator___Std;
+            public int m_Mean_Variance_Estimator___Gamma;
+            public int m_Mean_Variance_Estimator___Gamma_Initial_Mean;
+            public int m_Mean_Variance_Estimator___Gamma_Initial_Variance;
+            public int m_Mean_Variance_Estimator__Gamma_Mean;
+            public int m_Mean_Variance_Estimator__Gamma_Variance;
+            public int m_Mean_Variance_Estimator___Uptime_Gamma;
+            public int m_Mean_Variance_Estimator___Uptime_Gating;
+            public int m_Mean_Variance_Estimator___Gating_Duration_Minutes;
+            public int m_Mean_Variance_Estimator___Sigmoid__L;
+            public int m_Mean_Variance_Estimator___Sigmoid__K;
+            public int m_Mean_Variance_Estimator___Sigmoid__X0;
+            public int m_Mox_Model__Sraw_Std;
+            public int m_Mox_Model__Sraw_Mean;
+            public int m_Sigmoid_Scaled__Offset;
+            public int m_Adaptive_Lowpass__A1;
+            public int m_Adaptive_Lowpass__A2;
+            public bool m_Adaptive_Lowpass___Initialized;
+            public int m_Adaptive_Lowpass___X1;
+            public int m_Adaptive_Lowpass___X2;
+            public int m_Adaptive_Lowpass___X3;
+        };
+        /*
+ * Copyright (c) 2021, Sensirion AG
```

Comment:
There's a THIRD_PARTY_NOTICES file in root that this copyright can be moved to.

Reply:
Will add.